### ISSUE TITLE
6.0.1 contains a breaking change on `showRecordFields` ?

### ISSUE THREAD

**Issue by jmatsushita** [purescript/purescript-prelude#303]
PR #299 seems to contain a [breaking change](https://github.com/purescript/purescript-prelude/pull/299/files#diff-dd43a1d5c0d3cfc085d97d7d2f780b9aef9600e37e27959b9acf28d0ad2dede5L59) CC @ajnsit 

For instance it breaks [this code](https://github.com/ntwilson/purescript-option/blob/main/src/Option.purs#L439-L440):
```purescript
requiredFields :: Array String
requiredFields = Data.Show.showRecordFields requiredProxy (required record')
```

and returns this error:
```
[1/1 TypesDoNotUnify] .spago/option/main/src/Option.purs:440:22

  440      requiredFields = Data.Show.showRecordFields requiredProxy (required record')
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Could not match type

    String

  with type

    Array String
```

**Comment by garyb** [comment]
You're absolutely right, I considered `ShowRecordFields` to be an internal typeclass since it's a helper for the standard `Show` instance, forgetting that it has to be exported also, making it essentially public.

I'm not sure what the best course of action is now - to revert this, or to just fix `purescript-option` since it hasn't been reported anywhere else. I have to assume `purescript-option` is not in the registry as otherwise this would have become apparent sooner. 🤔

**Comment by JordanMartinez** [comment]
I think I agree that `ShowRecordFields` is an internal type class. I think `purescript-option` should have defined its own type class to handle its particular use case, but maybe they didn't because they would have just reimplemented `ShowRecordFields`... :thinking:
